synapse/lib/stormlib/easyperm.py

```diff
@@ -0,0 +1,93 @@
+
+            Note: The current user will be given admin permission in the new
+            easy perm structure.
+        ''',
+         'type': {'type': 'function', '_funcname': '_initEasyPerm',
+                  'args': (
+                      {'name': 'edef', 'type': 'dict', 'default': None,
+                       'desc': 'A dictionary to add easy perms to.'},
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'Dictionary with the easy perm structure.'}}},
+        {'name': 'set', 'desc': 'Set the permission level for a user or role in an easy perm dictionary.',
+         'type': {'type': 'function', '_funcname': '_setEasyPerm',
+                  'args': (
+                      {'name': 'edef', 'type': 'dict', 'desc': 'The easy perm dictionary to modify.'},
+                      {'name': 'scope', 'type': 'str', 'desc': 'The scope, either "users" or "roles".'},
+                      {'name': 'iden', 'type': 'str', 'desc': 'The user/role iden depending on scope.'},
+                      {'name': 'level', 'type': 'int', 'desc': 'The permission level number, or None to remove the permission.'},
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'Dictionary with the updated easy perm structure.'}}},
+        {'name': 'allowed', 'desc': 'Check if the current user has a permission level in an easy perm dictionary.',
+         'type': {'type': 'function', '_funcname': '_allowedEasyPerm',
+                  'args': (
+                      {'name': 'edef', 'type': 'dict', 'desc': 'The easy perm dictionary to check.'},
+                      {'name': 'level', 'type': 'str', 'desc': 'The required permission level number.'},
+                  ),
+                  'returns': {'type': 'boolean',
+                              'desc': 'True if the user meets the requirement, false otherwise.', }}},
+        {'name': 'confirm', 'desc': 'Require that the current user has a permission level in an easy perm dictionary.',
+         'type': {'type': 'function', '_funcname': '_confirmEasyPerm',
+                  'args': (
+                      {'name': 'edef', 'type': 'dict', 'desc': 'The easy perm dictionary to check.'},
+                      {'name': 'level', 'type': 'str', 'desc': 'The required permission level number.'},
+                  ),
+                  'returns': {'type': 'null'}}},
+    )
+    _storm_lib_path = ('auth', 'easyperm')
+
+    def getObjLocals(self):
+        return {
+            'set': self._setEasyPerm,
+            'init': self._initEasyPerm,
+            'allowed': self._allowedEasyPerm,
+            'confirm': self._confirmEasyPerm,
+            'level': {
+                'deny': s_cell.PERM_DENY,
+                'read': s_cell.PERM_READ,
+                'edit': s_cell.PERM_EDIT,
+                'admin': s_cell.PERM_ADMIN,
+            }
+        }
+
+    async def _setEasyPerm(self, edef, scope, iden, level):
+        edef = await s_stormtypes.toprim(edef)
```

Comment:
Can we add a isinstance check to ensure this is a `dict` to avoid a python error in _setEasyPerm ?